src/System.Text.Json/tests/Serialization/PropertyNameTests.cs

```diff
@@ -293,6 +293,87 @@ public static void ExtensionDataDictionarySerialize_DoesNotHonor()
             Assert.False(obj.MyOverflow.ContainsKey("key1"));
             Assert.Equal(1, obj.MyOverflow["Key1"].GetInt32());
         }
+
+        private class ClassWithPropertyNamePermutations
+        {
+            public int a { get; set; }
```

Comment:
Let's add the same test but for non-ascii property names.

Also, it would be good to add a test where the JsonPropertyName attribute has the names overriden (for all the same lengths, 1-10).

Reply:
Added the non-ascii properties.

Using the JsonPropertyName (and naming policy) is tested elsewhere and not related.